spec/datadog/core/process_discovery_spec.rb

```diff
@@ -85,4 +91,34 @@
       end
     end
   end
+
+  describe 'when forked', skip: !LibdatadogHelpers.supported? do
+    before do
+      Datadog.configure do |c|
+        c.service = 'test-service' # Manually set so it isn't set to fallback service name that we don't control
+      end
+    end
+
+    it 'updates the process discovery file descriptor' do
+      expect_in_fork do
+        described_class.get_and_store_metadata(Datadog.configuration, Datadog::Core::Logger.new(StringIO.new))
+        native_fd = described_class.instance_variable_get(:@file_descriptor)
+        fd = described_class._native_to_rb_int(native_fd)
+        buffer = IO.new(fd, autoclose: false)
+        buffer.rewind
+        content = MessagePack.unpack(buffer.read)
+
+        expect(content).to eq(
+          {
+            'schema_version' => 1,
+            'runtime_id' => Datadog::Core::Environment::Identity.id,
+            'tracer_language' => Datadog::Core::Environment::Identity.lang,
+            'tracer_version' => Datadog::Core::Environment::Identity.gem_datadog_version_semver2,
+            'hostname' => Datadog::Core::Environment::Socket.hostname,
+            'service_name' => 'test-service'
+          }
+        )
+      end
```

Comment:
This spec isn't very good: It passes if `apply_at_fork_patch` + the onlyonce get removed.

That's because it's testing the same codepath as above "hey if I call things manually they get called" but the codepath we need to check is "if there's a fork, do things get called automatically AND with the updated values.

Testing the atfork stuff is somewhat annoying. See https://github.com/DataDog/dd-trace-rb/pull/4204 for one example of how I tested it for crashtracking.